app/models/ability.rb

```diff
@@ -25,7 +32,21 @@ def publisher
   end
 
   def admin
+    raise AdminNotOnIPWhitelistError.new("Administrator must be IP whitelisted") unless admin_ip_whitelisted?
+    raise TwoFactorDisabledError.new("2fa must be enabled for administrators") unless two_factor_enabled?(@publisher)
     can :manage, :all
     can :access, :all
   end
+
+  def admin_ip_whitelisted?
+    return true if ADMIN_IP_WHITELIST.blank?
```

Comment:
this makes sense for dev, can we add an extra check against RAILS_ENV to return false if we're in production though?